nsxt/resource_nsxt_policy_transport_zone.go

```diff
@@ -0,0 +1,291 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/sites/enforcement_points"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+func resourceNsxtPolicyTransportZone() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyTransportZoneCreate,
+		Read:   resourceNsxtPolicyTransportZoneRead,
+		Update: resourceNsxtPolicyTransportZoneUpdate,
+		Delete: resourceNsxtPolicyTransportZoneDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceNsxtPolicyTransportZoneImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"display_name": getDataSourceDisplayNameSchema(),
+			"description":  getDataSourceDescriptionSchema(),
+			"path":         getPathSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"is_default": {
+				Type:        schema.TypeBool,
+				Description: "Indicates whether the transport zone is default",
+				Optional:    true,
+				Default:     false,
+			},
+			"transport_type": {
+				Type:         schema.TypeString,
+				Description:  "Type of Transport Zone",
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(policyTransportZoneTransportTypes, false),
+				ForceNew:     true,
+			},
+			"uplink_teaming_policy_names": {
+				Type:        schema.TypeList,
+				Description: "Names of the switching uplink teaming policies that are supported by this transport zone.",
+				Optional:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"site_id": {
```

Comment:
Lets consider changing this to `site_path` for consistency (we already have `site_path` in TZ data source and other resources)